### PR TITLE
Feature/tcvp 3030 springboot updates

### PR DESCRIPTION
**Pull request by jianmingtu**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jianmingtu — COMMENTED** [review]
Update Java from 11 to 17;  Springboot from 2.7.4 to 3.3.4
 Fixed the 24 high/critical vulnerabilities, see result, [vulnerabilities_after.txt](https://jag.gov.bc.ca/jira/secure/attachment/152079/152079_vulnerabilities_after.txt)
 Remove the H2 database from code base because we only use ORDS to fetch and update data
 Comment out the previous H2 unit tests because we are not connecting H2 database now, plus unit tests using H2 database are integration tests.
 Add these four files for unit testing EmailHistory in Controller, Reposition, Service and OutgoingEmailMapper.  This will be a sample patten if we need to unit test other ControllerReposition./Service/Mapper.
EmailHistory
├── EmailHistoryControllerTest.java
├── EmailHistoryRepositoryTest.java
├── EmailHistoryServiceTest.java
└── OutgoingEmailMapperTest.java

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeMapperTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeMapperTest.java:1]: You could have kept this mapper test as this is unrelated to H2 database.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapperTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapperTest.java:1]: We should keep this test as well.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapperTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapperTest.java:1]: We should keep this test as well.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/OutgoingEmailMapperTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/OutgoingEmailMapperTest.java:1]: We should keep this test as well.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java:1]: We should keep this test as well.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsOccamTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsOccamTest.java:1]: We can keep this test as well as it actually uses ORDS for creating the test records and deletes at the end of the test if the test is enabled.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsTcoTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsTcoTest.java:1]: Same for this test. It uses ORDS.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceTest.java:1]: We should keep this test as well.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/impl/ords/DisputeServiceImplTest.java
Comment (karahanb) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/impl/ords/DisputeServiceImplTest.java:1]: We should keep this test as well.

**Review by karahanb — CHANGES_REQUESTED** [review]
This is great work! All main functionalities work as expected with the upgraded libraries except the Redis connection when all services are started by docker compose. I get the following error: `org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis.` This might be a side effect of one of the updated libraries in pom since I could not see any code changes around the redis connection configuration or lookup service. 

Also, there are some useful test which are commented out could have kept since they are unrelated to H2 database and provide some unit and integration test coverage through either connecting to ORDS or mocking some services. I suggest to re-enable those tests that I individually commented on by isolating them from the H2 stuff in BaseTestSuite.

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeMapperTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeMapperTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapperTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapperTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsTcoTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsTcoTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapperTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapperTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/OutgoingEmailMapperTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/OutgoingEmailMapperTest.java:1]: reenabled

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsOccamTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceOrdsOccamTest.java:1]: DisputeServiceOrdsOccamTest of the previous springboot 2 version failed so  I could not reenable to the old version

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceTest.java
Comment (jianmingtu) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/DisputeServiceTest.java:1]: DisputeServiceOrdsTcoTest fails in the previous spring 2 so I could not reenable.

**Review by karahanb — APPROVED** [review]

**Pull request merged**